numba/cuda/cudadrv/devicearray.py

```diff
@@ -610,7 +610,8 @@ def _do_setitem(self, key, value, stream=0):
                 rhs.ndim,
                 lhs.ndim))
         rhs_shape = np.ones(lhs.ndim, dtype=np.int64)
-        rhs_shape[-rhs.ndim:] = rhs.shape
+        # negative indices would not work if rhs.ndim == 0
+        rhs_shape[lhs.ndim - rhs.ndim:] = rhs.shape
```

Comment:
@seibert, the changes to these lines should fix most of the failures in that traceback